.circleci/config.yml

```diff
@@ -55,49 +55,64 @@ jobs:
+          # meaning that both AWS Graviton instances and regular AMD
+          # instances can both use this image. Because both architectures need to be built,
+          # the build process will take longer than it historically has.
+
+          # the 'docker buildx create' command below creates a new builder instance that is capable
+          # of building multi-platform images, since the default instance does not
+          # have that capability by default.
+
+          # See https://www.docker.com/blog/faster-multi-platform-builds-dockerfile-cross-compilation-guide/
+          # for details
           command: |
-            mkdir -p /caches
-            docker save -o /caches/app.tar app
+            GIT_HASH=$(echo $CIRCLE_SHA1 | cut -c -7)
+            echo "$DOCKER_PASS" | docker login --username $DOCKER_USER --password-stdin
+            docker buildx create --name multi_platform_builder --use
+
+            docker buildx build \
+              --platform linux/amd64,linux/arm64 \
+              --cache-from type=local,src=/caches/docker-buildx \
+              --cache-to type=local,dest=/caches/docker-buildx,mode=max \
+              --tag acmucsd/membership-portal-api:latest \
+              --tag acmucsd/membership-portal-api:$GIT_HASH \
+              --push .
       - save_cache:
           key: v1-{{ .Branch }}-{{ epoch }}
           paths:
-            - /caches/app.tar
-      - run:
-          name: Push application Docker image
-          command: |
-            echo "$DOCKER_PASS" | docker login --username $DOCKER_USER --password-stdin
-            docker tag app "acmucsd/membership-portal-api:latest"
-            docker push "acmucsd/membership-portal-api:latest"
+            - /caches/docker-buildx
+
   deploy:
+    environment:
+      GIT_HASH: $(echo $CIRCLE_SHA1 | cut -c -7)
     machine:
       enabled: true
     steps:
       - checkout
       - run:
           command: |
-            docker run caprover/cli-caprover:v2.1.1 caprover deploy --caproverUrl https://captain.app.acmucsd.com --caproverPassword $CAPROVER_PASS --caproverApp membership-portal-api --imageName acmucsd/membership-portal-api:latest
-            docker run caprover/cli-caprover:v2.1.1 caprover deploy --caproverUrl https://captain.app.acmucsd.com --caproverPassword $CAPROVER_PASS --caproverApp membership-portal-api-testing --imageName acmucsd/membership-portal-api:latest
+            GIT_HASH=$(echo $CIRCLE_SHA1 | cut -c -7)
+            docker run caprover/cli-caprover:v2.1.1 caprover deploy --caproverUrl https://captain.caprover.acmucsd.com --caproverPassword $CAPROVER_PASS --caproverApp membership-portal-api-testing --imageName acmucsd/membership-portal-api:$GIT_HASH
+            docker run caprover/cli-caprover:v2.1.1 caprover deploy --caproverUrl https://captain.caprover.acmucsd.com --caproverPassword $CAPROVER_PASS --caproverApp membership-portal-api --imageName acmucsd/membership-portal-api:$GIT_HASH
 
 workflows:
   version: 2
```

Comment:
Is the workflow version supposed to differ from the config version?

Reply:
Good catch, seems like from [here](https://circleci.com/docs/configuration-reference/#workflow-version) the version field isn't required if the overall config version is 2.1, so I'll remove this